x-pack/test/functional/apps/ml/anomaly_detection_jobs/index.ts

```diff
@@ -8,40 +8,55 @@
 import { FtrProviderContext } from '../../../ftr_provider_context';
 
 export default function ({ getService, loadTestFile }: FtrProviderContext) {
-  const esArchiver = getService('esArchiver');
+  const config = getService('config');
+  const security = getService('security');
+  const isCcs = config.get('esTestCluster.ccs');
+  const esNode = isCcs ? getService('remoteEsArchiver' as 'esArchiver') : getService('esArchiver');
   const ml = getService('ml');
 
   describe('machine learning - anomaly detection', function () {
     this.tags(['skipFirefox']);
 
     before(async () => {
-      await ml.securityCommon.createMlRoles();
-      await ml.securityCommon.createMlUsers();
+      if (isCcs) {
+        await security.testUser.setRoles(['superuser']);
+      } else {
+        await ml.securityCommon.createMlRoles();
+        await ml.securityCommon.createMlUsers();
+        await ml.securityUI.loginAsMlPowerUser();
+      }
```

Comment:
I think it would be better to not change the existing login and user privileges logic for the CCS tests. Running the suite with superuser privileges makes us lose some test coverage. I'd suggest to modify our ML poweruser instead and add the remote role if required, e.g. like this: https://gist.github.com/pheyos/b4081f767d5e61e396b56604f250891f

Reply:
I disagree. I don't think that I should edit an existing user since I am not sure what the required perms is for the different tests. In the CCS test, we are not exercising tests on the permissions and whether or not data is being secured properly. We just want to make sure that CCS works on a remote cluster. I think having the security exercised on a local cluster is enough. But I could be wrong. What are your thoughts? @leedr @pheyos

Reply:
It's true that these tests are not specifically testing permissions on remote indices, but I think it's still valuable to run with a "regular" user here. In general, I try to avoid `superuser` usage in tests where possible. The reason is that this user is allowed to do (almost) everything so running tests with it could conceal an actual issue in the workflow. What if e.g. the ML app had an issue checking privileges on the selected remote data view? We wouldn't see that when running with a superuser.
And on the other hand, I don't see an issue with adding the remote privileges to our ML test user as [suggested](https://gist.github.com/pheyos/b4081f767d5e61e396b56604f250891f) (which are the same as are already [used](https://github.com/elastic/kibana/blob/main/x-pack/test/functional/config.ccs.ts#L42) for the internal test user to make the CCS tests work).

Reply:
@pheyos Fair enough. I will update and push. Thanks for the insight.